### PR TITLE
Fix notification list toolbar filter keys and convert Project/* to functional components

### PR DESCRIPTION
**Pull request by marshmalien**
##### SUMMARY
Issue: [https://github.com/ansible/awx/issues/8541](https://github.com/ansible/awx/issues/8541)

- In notification templates list `/next/notification_templates/`
    - Add "Description" to search options
    - Add "Type" to sort options
    - Fix notification type filtering bug
- In nested notification lists `/next/:resource/:resource_id/notifications`
    - Add notification "Type" filter options, "Created by", and "Modified by" to search options
- Convert Project.jsx, Projects.jsx, and ProjectSyncButton.jsx to functional components

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

 - UI

filter

### CODE REVIEW

File: awx/ui_next/src/screens/Project/shared/ProjectSyncButton.jsx [hunk 1]
```diff
@@ -1,70 +1,52 @@
...
+    <>
+      {children(handleSync)}
+      {error && (
+        <AlertModal
+          isOpen={error}
+          variant="error"
+          title={i18n._(t`Error!`)}
+          onClose={dismissError}
+        >
+          {i18n._(t`Failed to sync job.`)}
```
Comment (nixocio) [inline awx/ui_next/src/screens/Project/shared/ProjectSyncButton.jsx]:
```suggestion
          {i18n._(t`Failed to sync project.`)}
```

**Review by nixocio — COMMENTED** [review]
@marshmalien changes look pretty good. Small suggestion.

**Comment by nixocio** [comment]
Worth to run the e2e tests to verify we are adding any regression.

**Review by mabashian — APPROVED** [review]
lgtm

**Comment by mabashian** [comment]
@unlikelyzero @tiagodread This may be a good candidate for "run yolok and if it passes then merge".  I'll leave that up to y'all though.

**Comment by unlikelyzero** [comment]
@mabashian we will be looking at this one together to ensure we have the appropriate coverage. Back in June/July, we had some missed tests and this is potentially a blind spot

**Comment by tiagodread** [comment]
Hey @marshmalien Worked as expected filtering by type

Do you mind adding some uniqueness ids for the search toolbar elements like this:

image

**Comment by marshmalien** [comment]
@tiagodread I added `data-ouia-component-id="filter-by-${key}"` to the Select component and `id="select-option-${key}"` to the SelectOption checkbox items. Let me know if that helps!
Screen Shot 2020-11-30 at 11 55 36 AM

File: awx/ui_next/src/components/Search/Search.jsx [hunk 2]
```diff
@@ -234,6 +240,7 @@ function Search({
                 selections={chipsByKey[key].chips[0]?.label}
                 isOpen={isFilterDropdownOpen}
                 placeholderText={`Filter By ${name}`}
+                ouiaId={`filter-by-${key}`}
```
Comment (tiagodread) [inline awx/ui_next/src/components/Search/Search.jsx:243]: 🎉

**Comment by tiagodread** [comment]
@marshmalien I ran the ui tests and noticed that the edit operation broken:

proj

**Review by tiagodread — APPROVED** [review]
Verified that the user can filter notification by type, description, created and modified by on notification template list and notification tabs.

Verified the regression against projects, added coverage in 5738

**Pull request merged**